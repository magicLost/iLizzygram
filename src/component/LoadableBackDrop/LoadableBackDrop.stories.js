import React from "react";
import { action } from "@storybook/addon-actions";
import LoadableBackDrop from "./LoadableBackDrop";
        
export default {
    component: LoadableBackDrop,
    title: "LoadableBackDrop",
    decorators: [],
    //decorators: [withKnobs],
    // Our exports that end in "Data" are not stories.
    excludeStories: /.*Data$/,
};

export const Default = () => {
    return (
      <>
        <LoadableBackDrop />
      </>
    );
  };
        