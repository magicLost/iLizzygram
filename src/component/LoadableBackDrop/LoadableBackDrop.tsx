import React from "react";
import Backdrop from "@material-ui/core/Backdrop";
import CircularProgress from "@material-ui/core/CircularProgress";
import Button from "@material-ui/core/Button";
//import classes from './LoadableBackDrop.module.scss';
//import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface LoadableBackDropProps {}

/*const useStyles = makeStyles({
    root: {
      display: "block",
    },
});*/

const LoadableBackDrop = ({}: LoadableBackDropProps) => {
  //const classes = useStyles();

  const [open, setOpen] = React.useState(false);
  const handleClose = () => {
    setOpen(false);
  };
  const handleToggle = () => {
    setOpen(!open);
  };

  return (
    <>
      <Button variant="outlined" color="primary" onClick={handleToggle}>
        Show backdrop
      </Button>
      <Backdrop open={open} onClick={handleClose}>
        <CircularProgress color="inherit" />
      </Backdrop>
    </>
  );
};

export default LoadableBackDrop;
