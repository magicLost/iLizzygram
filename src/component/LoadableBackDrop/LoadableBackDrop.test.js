import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import LoadableBackDrop from "./LoadableBackDrop";
//import classes from "./LoadableBackDrop.module.scss";

describe("LoadableBackDrop", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {
      _render = render(<LoadableBackDrop />);
    });

    afterEach(cleanup);

    describe("Snapshots", () => {
      test("matches snapshot", () => {
        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
