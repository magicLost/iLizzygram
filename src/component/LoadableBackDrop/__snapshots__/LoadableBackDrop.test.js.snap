// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadableBackDrop Render and props test  matches snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Show backdrop
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 0; visibility: hidden;"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
